repo/incubating/elastic/versions/0/elastic-frameworkversion.yaml

```diff
@@ -0,0 +1,370 @@
+                  - name: discovery.seed_hosts
+                    value: {{NAME}}-master-0.{{NAME}}-master-hs,{{NAME}}-master-1.{{NAME}}-master-hs,{{NAME}}-master-2.{{NAME}}-master-hs
+                  - name: cluster.initial_master_nodes
+                    value: {{NAME}}-master-0,{{NAME}}-master-1,{{NAME}}-master-2
+                  - name: node.master
+                    value: "true"
+                  - name: node.data
+                    value: "false"
+                  - name: node.ingest
+                    value: "false"
+                  - name: cluster.remote.connect
+                    value: "false"
+                volumeMounts:
+                  - name: data
+                    mountPath: /usr/share/elasticsearch/data
+        volumeClaimTemplates:
+          - metadata:
+              name: data
+            spec:
+              accessModes: [ "ReadWriteOnce" ]
+              resources:
+                requests:
+                  storage: 1Gi
+    data-service.yaml: |
+      kind: Service
+      apiVersion: v1
+      metadata:
+        name: data-hs
+        namespace: {{NAMESPACE}}
+      spec:
+        selector:
+          app: data
+        ports:
+          - protocol: TCP
+            port: 9200
+        clusterIP: None
+    data.yaml: |
+      kind: StatefulSet
+      apiVersion: apps/v1
+      metadata:
+        name: data
+        namespace: {{NAMESPACE}}
+      spec:
+        selector:
+          matchLabels:
+            app: data # has to match .spec.template.metadata.labels
+        serviceName: {{NAME}}-data-hs
+        replicas: {{DATA_NODE_COUNT}}
+        template:
+          metadata:
+            labels:
+              app: data # has to match .spec.selector.matchLabels
+          spec:
+            initContainers:
+              - name: init-sysctl
+                image: busybox
+                command: ['sh', '-c', 'sysctl -w vm.max_map_count=262144']
+                securityContext:
+                  privileged: true
+              - name: volume-permissions
+                image: busybox
+                command: ['sh', '-c', 'chown -R 1000:1000 /usr/share/elasticsearch/data']
+                volumeMounts:
+                  - name: data
+                    mountPath: /usr/share/elasticsearch/data
+            terminationGracePeriodSeconds: 10
+            containers:
+              - name: elastic
+                image: elasticsearch:7.0.0
+                ports:
+                  - containerPort: 9200
+                    name: api
+                  - containerPort: 9300
+                    name: internal
+                env:
+                  - name: cluster.name
+                    value: {{NAME}}-cluster
+                  - name: discovery.seed_hosts
+                    value: {{NAME}}-master-0.{{NAME}}-master-hs,{{NAME}}-master-1.{{NAME}}-master-hs,{{NAME}}-master-2.{{NAME}}-master-hs
+                  - name: node.master
+                    value: "false"
```

Comment:
node.master, node.data, node.ingest configure the personality of the es node in this case we configure a data node, this can not be changed by a user.

cluster.remote.connect is for cross cluster search and cross cluster replication, we not getting into this at this point.